mage/algorithms/machine-learning-graph-analytics/node-classification-algorithm.md

```diff
@@ -5,22 +5,36 @@ sidebar_label: Node Classification
 ---
 
 ## Description
+Node classification is the problem of finding out the **right label** for a **node** based on its **neighbors’ labels** and **structure similarities**.
 
-Prediction can be done at the node level. The basis of such prediction systems are features extracted from graph entities.
+More formally, let `G=(V,E)` be a graph, `S` the set of all labels, and `f : V → S` a vertex labeling function that returns the correct label for each node. 
+From a given partial function `g : V’ ⊆ V → S` (`V’` being training examples), 
+a node classification model aims to minimize the loss of the label prediction function `g’ : V → S`.
 
-Extracting a feature can be a complicated problem, and it can be based on different graph attributes — node properties, node adjacency, or the structure of the neighborhood.
+The motivation for node classification models is the sociological concept of **homophily**. In social networks, individuals who associate and 
+bond with each other tend to have similar properties. In graph structure, **homophily** means that neighboring nodes 
+tend to have the same labels and similar connections.  
 
-Traditional methods of extracting knowledge from nodes include measures of [centrality](https://en.wikipedia.org/wiki/Centrality), importance, or feature structure such as [graphlets](https://en.wikipedia.org/wiki/Graphlets).
+Set `V` can be a lot of things, and the most common are scientific publications, proteins, Reddit posts, etc. In the same respective order, 
+set `E` can be a citation of publications, protein connections, or connection of posts commented by the same user. Therefore, node classification helps the user to classify publications 
+relevant to its field, check whether protein helps cure new diseases, or simply find out to which subreddit its post naturally belongs. 
+Given the not completely labeled graph, node classification tends to find the correct label for each node.
 
-Somewhat more advanced methods are extracting the [embedding](https://towardsdatascience.com/graph-embeddings-the-summary-cc6075aba007) of individual nodes, and then a prediction algorithm that takes knowledge from the embeddings themselves. The most popular such tool is [Node2Vec](/mage/algorithms/machine-learning-graph-analytics/node2vec-algorithm).
+### Solution 
+Solving methods range from **traditional** to **machine learning-based**. Both models work on top of the feature which describes each node. Together features and relationships between
```

Comment:
What do you mean by traditional here?

Reply:
Like label propagation algorithms, or something similar like that

Reply:
Thanks - I’d replace traditional with algorithmic here